pkg/scheduler/actions/allocate/allocate.go

```diff
@@ -98,11 +98,16 @@ func (alloc *Action) Execute(ssn *framework.Session) {
 	allNodes := ssn.NodeList
 	predicateFn := func(task *api.TaskInfo, node *api.NodeInfo) error {
 		// Check for Resource Predicate
-		if ok, reason := task.InitResreq.LessEqualWithReason(node.FutureIdle(), api.Zero); !ok {
-			return api.NewFitError(task, node, reason)
+		if err := ssn.PredicateResourceFn(task, node); err != nil {
```

Comment:
PredicateResource is just a resource comparation and do we have to make it as a session API. 
It seems there is no need to implement the function with different ways in different plugins.


make it as a session API.